### PR TITLE
[Doc] Improve multiple LightningTrainer Documentations

### LINKED ISSUE #35626: [Doc] Separating the constant configs and searchable configs in the LightningTrainer + Tune Example.
### Description

Users reported that some objects cannot be passed through search_space. (#35571) We should explicitly separate the definition of searchable configs and constant configs in our example. Currently they are mixed together which mislead the users.

### Link

_No response_

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Improve multiple docs for LightningTrainer based on recent users' feedbacks:

- Improved the Tuner + LightningTrainer example
  - Replace the deprecated method `validation_epoch_end` with `on_validation_epoch_end`.
  - Separate the `lightning_configs` into static configs and searchable configs. Use one for building the base `LightningTrainer` and one for defining the tuner `param_space`.
- Added a new line in the trainer catalog table (`train.rst`) for `LightningTrainer`
- Specified `run_id` to avoid duplicated runs in LightningTrainer + Experiment Tracking Tools example.
- Explicitly call `wandb.finish()` in WandbLogger to ensure all artifacts are uploaded before training finished.

## Related issue number

Closes #35626 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/examples/lightning/lightning_exp_tracking.ipynb [hunk 1]
```diff
@@ -131,9 +131,18 @@
...
     "    # Wandb\n",
     "    wandb_api_key = os.environ.get(\"WANDB_API_KEY\", None)\n",
-    "    wandb_logger = WandbLogger(\n",
+    "\n",
+    "    class RayWandbLogger(WandbLogger):\n",
+    "        # wandb.finish() ensures all artifacts get uploaded at the end of training.\n",
+    "        def finalize(self, status):\n",
+    "            super().finalize(status)\n",
+    "            wandb.finish()\n",
+    "\n",
```
Comment (richardliaw) [inline doc/source/train/examples/lightning/lightning_exp_tracking.ipynb:140]: why don't we do this by default in the wandblogger?

**Review by richardliaw — APPROVED** [review]
changes look good, just 1 question about why we need to subclass

File: doc/source/train/examples/lightning/lightning_exp_tracking.ipynb (hunk 1 above)
Comment (woshiyyya) [reply doc/source/train/examples/lightning/lightning_exp_tracking.ipynb:140]: WandbLogger is a PyTorch lightning module. I'll create an issue to their repo.

More context here:
For performance concern, wandb asynchronously uploads artifacts and automatically triggers `wandb.finish()`(as a barrier) when your script exits, so PyTorch Lightning don't do this explicitly.

In Ray, we kill actors right after training is complete, so wandb may not be able to upload all files before the actor dies. So we have to explicitly call `finish()`.

Comment (richardliaw) [reply doc/source/train/examples/lightning/lightning_exp_tracking.ipynb:140]: nice!

**Pull request merged**